mygit.rb

```diff
@@ -0,0 +1,365 @@
+        else
+                system "git push -u origin #{directory}"
+        end
+        $LOG.debug("<MyGit> <mygit_prepare> : git pushed #{directory}")
+end
+
+def mygit_request(reviewers)
+        $LOG.debug("Request Called")
+        $LOG.debug("<MYGIT> <mygit_request>module : Check if git exists? : #{GitExist?()}")
+        reviewers.shift
+        repo_directory = File.open("repo.txt")
+        repo_directory = repo_directory.read
+        repo = getreponame()
+        
+        Dir.chdir(repo)
+        
+        base_branch = readhidden(repo,"Base.txt")       
+
+        working_branch = readhidden(repo,"Direx.txt")
+
+	Dir.chdir('..')
+        client = octoKitInit()
+
+        create_pull = client.create_pull_request(repo_directory,base_branch, working_branch , "From Girish's App", "Pull Request body")
+        
+        reqreview=client.request_pull_request_review(repo_directory, create_pull.number , reviewers: reviewers)
+        
+        reviewers.each do |reviewer|        
+                puts "<MyGit> : Review requested by #{reviewer}"
+        end
+
+
+        comparison_branches = client.compare(repo_directory, base_branch ,working_branch , options = {})
+        comparison_json = Hash.new
+        counter=0
+        comparison_branches.files.each do |file|
+                array_files=[]  
+
+                file.each do |f|
+                        array_files.push(f)
+        
+                end
+
+                comparison_json[counter] = array_files
+                        counter=counter+1
+        end
+        File.open("comparison_result.json","w") do |f|
+                f.write(comparison_json.to_json)
+        end
+
+        writetofile("pullreq.txt", create_pull.number, repo)
+        writetofile("reviewreq.txt", reqreview.id , repo)                        
+  
+                                
+end
+
+def mygit_status()
+        client = octoKitInit()
+        repo = getreponame()
+        repo_directory = File.open("repo.txt").read
+        Dir.chdir(repo)
+        pullreq=readhidden(repo,"pullreq.txt")
+        reviewreq = readhidden(repo,"reviewreq.txt")
+        base = readhidden(repo,"Base.txt")
+
+        review_status=client.pull_request_reviews(repo_directory, pullreq)
+                        
+        if review_status.empty?
+                 puts "<MyGit> : Review not done yet"
+                 exit 
+        else
+                review_ix= review_status.length - 1
+                
+                status=client.pull_request_review(repo_directory, pullreq, review_status[review_ix].id)
+
+                                                                                                                
+                puts status.state
+
+        end
+
+                
+        if status.state == 'APPROVED' 
+                puts "<MyGit> : your review is approved"
+                system "git add ."
+                system "git commit -m 'from Girish's App'"
```

Comment:
Commit message should be a param from the user.

Must allow multiline commit message.